staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -297,7 +297,7 @@ func (qs *queueSet) StartRequest(ctx context.Context, hashValue uint64, flowDist
 			// known that the count does not need to be accurate.
 			// BTW, the count only needs to be accurate in a test that
 			// uses FakeEventClock::Run().
-			klog.V(6).Infof("QS(%s): Context of request %q %#+v %#+v is Done", qs.qCfg.Name, fsName, descr1, descr2)
+			klog.V(6).Infof("Context of request %q %#+v %#+v is Done", fsName, descr1, descr2)
```

Comment:
@MikeSpreitzer I don't think it's worth acquiring a lock to get the queset name. This is a quick fix to unblock the release. Also, I will scan if there are other places where we print `qCfg.Name` of the queueset  that is not protected.

Reply:
you could grab the name outside the goroutine into a local var while under lock

Reply:
also, it looks like this was the same in 1.20, so while important to fix (and backport), it is not a regression?

 https://github.com/kubernetes/kubernetes/blob/release-1.20/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go#L300

Reply:
@liggitt we added this test in 1.21: https://github.com/kubernetes/kubernetes/pull/97471. I don't think it's a regression, the unprotected read of the queueset config name was always there and this new test has a certain setup that allows the data race to manifest, I have tried to explain this below: 

This data race manifests via this test because:
- this test exercises a request that is queued (the goroutine that is reading the queueset config name without a lock)
- this test uses the apf config controller which reads the configuration objects and updates the queueset states (the goroutine that writes the queueset internal states using a lock)

we don't have `status` reporting of `prioritylevelconfiguration` yet, which a test can reliably wait on to indicate that the controller has finished updating the queueset states. due to the lack of this feature, today p&f tests wait on the corresponding metric to be recorded appropriately. 
p&f config controller updates the metrics first and then changes the internal queueset states, so this gap in the wait mechanism used by the test leads to the datarace.